sentinel-dashboard/src/main/java/com/alibaba/csp/sentinel/dashboard/datasource/entity/rule/ParamFlowRuleEntity.java

```diff
@@ -85,9 +105,9 @@ public boolean isClusterMode() {
     public ParamFlowClusterConfig getClusterConfig() {
         return rule.getClusterConfig();
     }
-    
+
     @Override
-    public Rule toRule() {
+    public ParamFlowRule toRule() {
```

Comment:
Or we can move it up to `AbstractRuleEntity` using generic type?

Reply:
And maybe the same to `getLimitApp()` / `getResouce`

Reply:
> Or we can move it up to `AbstractRuleEntity` using generic type?

Good idea. I've polished the code.

> And maybe the same to `getLimitApp()` / `getResouce`

In fact, for some rules (e.g. `SystemRule`), `resource` and `limitApp` make no sense. We could even consider removing the `AbstractRule` in later versions (rules can be derived from a common empty `Rule` interface, but the fields of every rule type would be individual).

Reply:
Yes it can be reasonable if so. And indeed fastjson's bug will not hit so much after that.

And i quoted you several times in the fastjson issue and please give a response or your conclusion if you have time. Mainly about the injected location and prompting method.

Reply:
I'll reply in #736 soon.

For this PR, if it's okay, you could submit an "approve" review with comment "LGTM".